lib/online_order.rb

```diff
@@ -0,0 +1,97 @@
+require "csv"
+require_relative "order"
+require_relative "customer"
+
+ONLINEORDERS = "support/online_orders.csv"
+
+module Grocery
+  class OnlineOrder < Grocery::Order
+    attr_reader :id, :products, :customer_id, :status
+    def initialize(id, products, customer_id, status)
+      super(id, products)
+      @customer_id = customer_id.to_i
+      @status = status
+    end
+
+    # Modify total method to include shipping fee
+    def total
+      shipping_fee = 10
+      if super != 0
+        return super + shipping_fee
+      else
+        return super
+      end
+    end
+
+    # Modify add_method from parent class
+    # Only permit one new product to be added when status is pending or paid
+    def add_product(product_name, product_price)
+      if @status == "pending" || @status == "paid"
+        if !@products.keys.include?(product_name)
+          @products[product_name] = product_price
+        end
+      else
+        ArgumentError
+      end
+    end
+
+    # return an array of OnlineOrder s
+    def self.all
+      online_orders = []
+      CSV.read(ONLINEORDERS, "r").each do |online_order|
+        id = online_order[0].to_i
+        orders = online_order[1].split(%r{;\s*})
+        order_hash = {}
+        orders.each do |order|
+          pair = order.split(%r{:\s*})
```

Comment:
I would recommend against using the `%r{...}` syntax for regular expressions. `/;\s*/` is a more common way to write it, and it matches up with the syntax used by the program [sed](https://www.gnu.org/software/sed/manual/sed.html) (super useful but slightly arcane command line utility to do regex search/replace)

Reply:
Changed regex. Thanks for the useful reference.